commands/upgrade/upgrade_path.py

```diff
@@ -0,0 +1,124 @@
+import json
+import os
+
+from leapp.exceptions import CommandError
+
+HANA_BASE_PATH = '/hana/shared'
+HANA_SAPCONTROL_PATH = 'exe/linuxx86_64/hdb/sapcontrol'
+
+LEAPP_UPGRADE_FLAVOUR_DEFAULT = 'default'
+LEAPP_UPGRADE_FLAVOUR_SAP_HANA = 'saphana'
+LEAPP_UPGRADE_PATHS = 'upgrade_paths.json'
+
+
+# NOTE(ivasilev) Shamelessly copied from leapp.libraries.common.config as could not include shared actor
+# library at the command level
+def get_major_version(version):
+    """
+    Return the major version from the given version string.
+
+    Versioning schema: MAJOR.MINOR.PATCH
+    It doesn't matter how many dots are present. Everything until the first dot is returned. E.g.:
+       8.1.0 => 8
+       7.9   => 7
+       7     => 7
+
+    :param str version: The version string according to the versioning schema described.
+    :rtype: str
+    :returns: The major version from the given version string.
+    """
+    return version.split('.')[0]
+
+
+def detect_sap_hana():
+    """
+    Detect SAP HANA based on existance of /hana/shared/*/exe/linuxx86_64/hdb/sapcontrol
+    """
+    if os.path.exists(HANA_BASE_PATH):
+        for entry in os.listdir(HANA_BASE_PATH):
+            # Does /hana/shared/{entry}/exe/linuxx86_64/hdb/sapcontrol exist?
+            if os.path.exists(os.path.join(HANA_BASE_PATH, entry, HANA_SAPCONTROL_PATH)):
+                return True
+    return False
+
+
+def get_upgrade_flavour():
+    """
+    Returns the flavour of the upgrade for this system.
+    """
+    if detect_sap_hana():
+        return LEAPP_UPGRADE_FLAVOUR_SAP_HANA
+    return LEAPP_UPGRADE_FLAVOUR_DEFAULT
+
+
+def get_os_release_version_id(filepath):
+    """
+    Retrieve data about System OS release from provided file.
+
+    :return: `str` version_id
```

Comment:
@fernflower  we should maybe consolidate rules about docstrings :) currently we are using 3 docstring formats around our projects. in this file we are using two. maybe good topic for the next year :D

Reply:
And what's wrong with this one, you mean :return: in code?